flight/targets/playuavosd/fw/Makefile

```diff
@@ -0,0 +1,326 @@
+#
+# Additional note on redistribution: The copyright and license notices above
+# must be maintained in each individual source file that is a derivative work
+# of this source file; otherwise redistribution is prohibited.
+
+include $(MAKE_INC_DIR)/firmware-defs.mk
+include $(BOARD_INFO_DIR)/board-info.mk
+
+# Set developer code and compile options
+# Set to YES for debugging
+DEBUG ?= NO
+
+# List of modules to include
+MODULES  = FirmwareIAP
+MODULES += Telemetry
+
+OPTMODULES += GPS
+OPTMODULES += FlightStats
+OPTMODULES += TxPID
+OPTMODULES += ComUsbBridge
+#OPTMODULES += PicoC
+OPTMODULES += Logging
+OPTMODULES += VTXConfig
+OPTMODULES += OnScreenDisplay
+OPTMODULES += MSPUAVOBridge
+
+# Paths
+OPUAVTALKINC = $(OPUAVTALK)/inc
+OPUAVOBJINC = $(OPUAVOBJ)/inc
+PIOSINC = $(PIOS)/inc
+FLIGHTLIBINC = $(FLIGHTLIB)/inc
+MATHLIB = $(FLIGHTLIB)/math
+MATHLIBINC = $(FLIGHTLIB)/math
+MGRSLIB = $(FLIGHTLIB)/mgrs
+MGRSLIBINC = $(FLIGHTLIB)/mgrs
+RSCODE = $(FLIGHTLIB)/rscode
+RSCODEINC = $(FLIGHTLIB)/rscode
+PIOSSTM32F4XX = $(PIOS)/STM32F4xx
+PIOSCOMMON = $(PIOS)/Common
+PIOSCOMMONLIB = $(PIOSCOMMON)/Libraries
+APPLIBDIR = $(PIOSSTM32F4XX)/Libraries
+STMLIBDIR = $(APPLIBDIR)
+STMSPDDIR = $(STMLIBDIR)/STM32F4xx_StdPeriph_Driver
+STMSPDSRCDIR = $(STMSPDDIR)/src
+STMSPDINCDIR = $(STMSPDDIR)/inc
+DEBUG_CM3_DIR = $(PIOSCOMMONLIB)/Debug
+DEBUG_CM3_DIR_INC = $(DEBUG_CM3_DIR)/inc
+OPUAVOBJINC = $(OPUAVOBJ)/inc
+FLIGHTPLANLIB = $(OPMODULEDIR)/FlightPlan/lib
+FLIGHTPLANS = $(OPMODULEDIR)/FlightPlan/flightplans
+MAVLINKINC = $(FLIGHTLIB)/mavlink/v1.0/common
+MSPLIB = $(FLIGHTLIB)/msp
+
+# optional component libraries
+include $(APPLIBDIR)/ChibiOS/library.mk
+
+# List C source files here. (C dependencies are automatically generated.)
+# use file-extension c for "c-only"-files
+
+## MODULES
+SRC += ${foreach MOD, ${MODULES} ${OPTMODULES}, ${wildcard ${OPMODULEDIR}/${MOD}/*.c}}
+## OPENPILOT CORE:
+SRC += ${OPMODULEDIR}/System/systemmod.c
+SRC += main.c
+SRC += board.c
+SRC += pios_board.c
+SRC += pios_usb_board_data.c
+SRC += $(FLIGHTLIB)/alarms.c
+SRC += $(OPUAVTALK)/uavtalk.c
+SRC += $(OPUAVOBJ)/uavobjectmanager.c
+
+ifeq ($(DEBUG),YES)
+SRC += $(DEBUG_CM3_DIR)/dcc_stdio.c
+SRC += $(DEBUG_CM3_DIR)/cm3_fault_handlers.c
+endif
+
+SRC += $(FLIGHTLIB)/paths.c
+SRC += $(FLIGHTLIB)/WorldMagModel.c
+SRC += $(FLIGHTLIB)/insgps14state.c
+SRC += $(FLIGHTLIB)/taskmonitor.c
+SRC += $(FLIGHTLIB)/sanitycheck.c
+SRC += $(FLIGHTLIB)/timeutils.c
+SRC += $(FLIGHTLIB)/circqueue.c
+SRC += $(MATHLIB)/coordinate_conversions.c
+SRC += $(MATHLIB)/misc_math.c
+SRC += $(MATHLIB)/pid.c
+SRC += $(MATHLIB)/atmospheric_math.c
+
+## MGRS Library (needed by OSD)
+SRC += $(MGRSLIB)/mgrs.c
+SRC += $(MGRSLIB)/utm.c
+SRC += $(MGRSLIB)/ups.c
+SRC += $(MGRSLIB)/tranmerc.c
+SRC += $(MGRSLIB)/polarst.c
+
+# MSP protocol lib
+SRC += $(MSPLIB)/msp.c
+
+## PIOS Hardware (STM32F4xx)
+include $(PIOS)/STM32F4xx/library_chibios.mk
+# SRC += $(PIOS)/STM32F4xx/pios_video.c
+
+## PIOS Hardware (Common)
```

Comment:
Maybe, running a receiver into the OSD is not that crazy of a thing to do though, esp. if you have separate camera operator + pilot :P, and also might be desirable for some firmwares depending on available telemetry messages. ATM they are unused though.